metricbeat/module/kubernetes/container/container.go

```diff
@@ -71,25 +73,33 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}, nil
 }
 
-// Fetch methods implements the data gathering and data conversion to the right format
-// It returns the event which is then forward to the output. In case of an error, a
-// descriptive error must be returned.
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 	m.enricher.Start()
 
 	body, err := m.http.FetchContent()
 	if err != nil {
-		return nil, err
+		logger.Error(err)
+		reporter.Error(err)
+		return
 	}
 
 	events, err := eventMapping(body, util.PerfMetrics)
 	if err != nil {
-		return nil, err
+		logger.Error(err)
```

Comment:
should we add a err= errors.wrap(err, "eventMapping failed") here to give more content?

Reply:
Good question. I'm not sure about it because I usually like to print errors that are relevant for the users (not for us) to help them fix their config issues or similar things.

But I'm also sure that I'm not probably consistent about this :smile: